File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxltnrn.java
// EXPECTED: 90.50
// Reference emit for IDXLTNRN — INDEXED START LESS + READ NEXT.
import java.util.*;
public final class IdxltnrnRef {
  public static void main(String[] args) {
    Map<Integer, Double> rows = new HashMap<>();
    rows.put(10, 12.50); rows.put(20, 8.00); rows.put(42, 25.00);
    rows.put(55, 30.00); rows.put(99, 15.00);
    int start = 20;
    Integer pos = null;
    for (int k : rows.keySet()) {
      if (k < start && (pos == null || k > pos)) pos = k;
    }
    double total = 0;
    if (pos != null) {
      for (Map.Entry<Integer, Double> e : rows.entrySet()) {
        if (e.getKey() >= pos) total += e.getValue();
      }
    }
    System.out.printf(Locale.US, "%.2f%n", total);
  }
}

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxltnrn.py
"""Reference emit for IDXLTNRN — INDEXED START LESS + READ NEXT → 90.50."""
# EXPECTED: 90.50
ROWS = {10: 12.50, 20: 8.00, 42: 25.00, 55: 30.00, 99: 15.00}
START = 20
# GnuCOBOL BDB: START LESS positions at greatest key < START; NEXT includes it.
pos = max((int(k) for k in ROWS if int(k) < START), default=None)
total = sum(v for k, v in ROWS.items() if pos is not None and int(k) >= pos)
print(f"{total:.2f}")
